modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/system/test/AddressSystemObjectDefinitionManagerTest.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.internal.system.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.object.test.util.BaseSystemObjectDefinitionManagerTestCase;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.Address;
+import com.liferay.portal.kernel.model.BaseModel;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.test.AssertUtils;
+import com.liferay.portal.kernel.test.TestInfo;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Mario Gomes
+ */
+@RunWith(Arquillian.class)
+public class AddressSystemObjectDefinitionManagerTest
+	extends BaseSystemObjectDefinitionManagerTestCase {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	@Override
+	public void setUp() throws Exception {
+		super.setUp();
+	}
+
+	@After
+	@Override
+	public void tearDown() throws Exception {
+		super.tearDown();
+	}
+
+	@Override
+	@Test
+	@TestInfo("LPD-63933")
+	public void testGetOrAddEmptyBaseModel() throws Exception {
+		super.testGetOrAddEmptyBaseModel();
+	}
+
+	@Override
+	protected void assertFailureWithoutPermission(
+		BaseModel<?> baseModel, User user1, User user2) {
+
+		AssertUtils.assertFailure(
+			PortalException.class,
+			StringBundler.concat(
+				"User ", user2.getUserId(), " must have UPDATE permission for ",
+				User.class.getName(), " ", user1.getUserId()),
+			() -> systemObjectDefinitionManager.getOrAddEmptyBaseModel(
+				RandomTestUtil.randomString(), user1));
```

Comment:
this is strange because we just set the `user2` before executing this method
```
User user2 = UserTestUtil.addUser();

setUser(user2);
```

`user2` should be the one executing this action:
```
() -> systemObjectDefinitionManager.getOrAddEmptyBaseModel(
				RandomTestUtil.randomString(), user1)
```